go-controller/pkg/ovn/master.go

```diff
@@ -439,12 +440,12 @@ func (oc *Controller) syncGatewayLogicalNetwork(node *kapi.Node, l3GatewayConfig
 		return err
 	}
 
-	err = gatewayInit(node.Name, clusterSubnets, hostSubnets, joinSubnets, l3GatewayConfig, oc.SCTPSupport)
+	err = gatewayInit(node.Name, clusterSubnets, hostSubnets, joinSubnets, l3GatewayConfig, hybridGatewayConfig, oc.SCTPSupport)
 	if err != nil {
 		return fmt.Errorf("failed to init shared interface gateway: %v", err)
 	}
 
-	if l3GatewayConfig.Mode == config.GatewayModeShared {
+	if l3GatewayConfig.Mode == config.GatewayModeShared || config.Gateway.Mode == config.GatewayModeHybrid {
```

Comment:
This is wrong. The master should not care about config.Gateway.Mode; it must only look at that mode the nodes post themselves in their annotation.

Reply:
this is a more correct check since then we dont assume the node is the same gateway mode as the master, however this is assumed in the previous part of this expression with "l3GatewayConfig.Mode == config.GatewayModeShared". Either way I'll change it as you suggested.

Reply:
Yeah, but then we need to have a larger conversation with upstream/Girish/etc about no longer supporting mixed gateway mode in clusters...

Reply:
OK I'll bring it up in next community meeting.

Reply:
"supporting mixed gateway mode" is potentially important for eventual hybrid->shared migration...